### PR TITLE
Code Prettifying

### PR DESCRIPTION
**Pull request by Saintis**
Major attempt to make use of some (hopefully) more readable functions and idioms.

Does the following:
- Adds and makes use of `in-hand?` `is-type?` `has-subtype?`
- Makes use of `installed?` and `ice?` more extensive
- Splits several `and` and `or` expressions into several lines for readability
- Changes occurrences of `(if (not x))` to `(if-not x)`, same for `when`
- Makes use of `when` instead of `if` when relevant
- Makes use of `pos?` and `neg?`
- Refactors `resolve-ability` into several helper functions
- Removes unused `side-str` function
- Adds some docstrings
- Creates `card-name` that replaces `(if (:rezzed card) (:title card) "a card")` found throughout

Probably needs a lot of review as there are a lot of changes :smile:

### CODE REVIEW

File: src/clj/game/cards-resources.clj [hunk 1]
```diff
@@ -151,9 +152,9 @@
    "Eden Shard"
    {:abilities [{:effect (effect (trash card {:cause :ability-cost}) (draw :corp 2))
                  :msg "force the Corp to draw 2 cards"}]
-    :install-cost-bonus (req (if (and run (= (:server run) [:rd]) (= 0 (:position run)))
+    :install-cost-bonus (req (if (and run (= (:server run) [:rd]) (zero? (:position run)))
```
Comment (zaroth) [inline src/clj/game/cards-resources.clj]: I take it the `(and run` part takes care about this `zero?` having `nil` issues you talked about?

File: src/clj/game/core-abilities.clj [hunk 2]
```diff
@@ -135,7 +209,7 @@
                                                     (resolve-ability state side no-ability card targets)))))
 
 
-; Prompts
+;;; Prompts
```
Comment (zaroth) [inline src/clj/game/core-abilities.clj]: so now we're going triple semicolon even? ;-)

File: src/clj/game/core-actions.clj [hunk 3]
```diff
@@ -57,27 +57,31 @@
...
+                                 (= last-zone :hand)
+                                 (= server "Grip")))
+                       (or (and (= (:side c)  "Runner")
+                                (not (:facedown c)))
+                           (rezzed? c)
                            (:seen c)
                            (= last-zone :deck)))
-                (:title c) "a card")
+                (:title c)
+                "a card")
```
Comment (zaroth) [inline src/clj/game/core-actions.clj]: the whole naming in here just asks to be replaced by an appropriate `card-str` call, maybe modified slightly. Ideally all card naming, if not very specialized for some reason, could be handled in that one place.

**Comment by zaroth** [comment]
Great job! :+1: Had a few tiny concerns/comments, added them in line.

File: src/clj/game/cards-resources.clj (hunk 1 above)
Comment (Saintis) [inline src/clj/game/cards-resources.clj]: Didn't realise the `nil` issues until after this I think. Will probably revert this change to be sure.

File: src/clj/game/core-actions.clj (hunk 3 above)
Comment (nealterrell) [inline src/clj/game/core-actions.clj]: :+1: to what @zaroth said about `card-str`.

**Comment by Saintis** [comment]
Have fixed up the `card-str` stuff. Is ready for merge.

**Comment by mtgred** [comment]
There are merge conflicts. Can you have a look?

**Comment by Saintis** [comment]
Should work now (may have the test failure issue solved in #1104)

**Comment by Saintis** [comment]
Rebased changes on the fixes in #1104.

**Comment by mtgred** [comment]
There are merge conflicts. Can you have a look?

**Comment by Saintis** [comment]
There will always be merge conflicts :wink:
Fixed now.

**Comment by mtgred** [comment]
:+1:
